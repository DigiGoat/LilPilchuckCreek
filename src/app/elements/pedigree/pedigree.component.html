<div class="text-center mt-4">
  <div class="container border rounded-4 desktop-grid" *ngIf="goat.damId || goat.sireId">
    <div class="row row-cols-2 border-bottom">
      <span class="col border-end border-3 fw-bolder">Dam:</span>
      <span class="col fw-bolder">Sire:</span>
    </div>
    <div class="row row-cols-2 border-bottom border-3" *ngIf="dam || sire">
      <span class="col border-end border-3">{{dam?.name}}</span>
      <span class="col">{{sire?.name}}</span>
    </div>
    <div class="row row-cols-2 border-bottom border-3" *ngIf="!dam && !sire">
      <span class="col border-end border-3 placeholder-wave placeholder">lorem<br><small>ipsum</small></span>
      <span class="col placeholder-wave placeholder">lorem<br><small>ipsum</small></span>
    </div>
    <div class="row row-cols-4 border-bottom">
      <span class="col fw-bolder border-end border-3">Dam's Dam:</span>
      <span class="col fw-bolder border-end border-5">Dam's Sire:</span>
      <span class="col fw-bolder border-end border-3">Sire's Dam:</span>
      <span class="col fw-bolder ">Sire's Sire:</span>
    </div>
    <div class="row row-cols-4" *ngIf="damDam || damSire || sireDam || sireSire">
      <span class="col border-end border-3">{{damDam?.name}}</span>
      <span class="col border-end border-5">{{damSire?.name}}</span>
      <span class="col border-end border-3">{{sireDam?.name}}</span>
      <span class="col">{{sireSire?.name}}</span>
    </div>
    <div class="row row-cols-4" *ngIf="!damDam && !damSire && !sireDam && !sireSire">
      <span class="col border-end border-3 placeholder-wave placeholder">lorem <br><small>ipsum</small>
      </span>
      <span class="col border-end border-5 placeholder-wave placeholder">lorem <br><small>ipsum</small>
      </span>
      <span class="col border-end border-3 placeholder-wave placeholder">lorem<br><small>ipsum</small>
      </span>
      <span class="col placeholder placeholder-wave">lorem <br><small>ipsum</small>
      </span>
    </div>
  </div>
  <div class="container border rounded-4 mobile-grid" *ngIf="goat.damId">
    <div class="row row-cols-1 border-bottom">
      <span class="col">Dam:</span>
    </div>
    <div class="row row-cols-1 border-bottom border-3" *ngIf="dam">
      <span class="col">{{dam.name}}</span>
    </div>
    <div class="row row-cols-1 border-bottom border-3" *ngIf="!dam">
      <span class="col placeholder-wave placeholder">lorem<br><small>ipsum</small></span>
    </div>
    <div class="row row-cols-2 border-bottom">
      <span class="colborder-end border-3">Dam's Dam:</span>
      <span class="col ">Dam's Sire:</span>
    </div>
    <div class="row row-cols-2" *ngIf="damDam || damSire">
      <span class="col border-end border-3">{{damDam?.name}}</span>
      <span class="col">{{damSire?.name}} </span>
    </div>
    <div class="row row-cols-2" *ngIf="!damDam && !damSire">
      <span class="col border-end border-3 placeholder-wave placeholder">lorem<br><small>ipsum</small>
      </span>
      <span class="col placeholder-wave placeholder">lorem<br><small>ipsum</small>
      </span>
    </div>
  </div>
  <br *ngIf="goat.damId && goat.sireId">
  <div class="container border rounded-4 mobile-grid" *ngIf="goat.sireId">
    <div class="row row-cols-1 border-bottom">
      <span class="col">Sire:</span>
    </div>
    <div class="row row-cols-1 border-bottom border-3" *ngIf="sire">
      <span class="col">{{sire.name}}</span>
    </div>
    <div class="row row-cols-1 border-bottom border-3" *ngIf="!sire">
      <span class="col placeholder-wave placeholder">lorem<br><small>ipsum</small></span>
    </div>
    <div class="row row-cols-2 border-bottom">
      <span class="col border-end border-3">Sire's Dam:</span>
      <span class="col">Sire's Sire:</span>
    </div>
    <div class="row row-cols-2" *ngIf="sireDam || sireSire">
      <span class="col border-end border-3">{{sireDam?.name}} </span>
      <span class="col">{{sireSire?.name}} </span>
    </div>
    <div class="row row-cols-2" *ngIf="!sireDam && !sireSire">
      <span class="col border-end border-3 placeholder-wave placeholder">lorem<br><small>ipsum</small>
      </span>
      <span class="col placeholder-wave placeholder">lorem<br><small>ipsum</small>
      </span>
    </div>
  </div>
</div>