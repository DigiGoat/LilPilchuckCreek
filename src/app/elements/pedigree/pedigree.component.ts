import { Component, Input, type OnInit } from '@angular/core';
import { type Goat, GoatService } from '../../services/goat/goat.service';

@Component({
  selector: 'app-pedigree',
  templateUrl: './pedigree.component.html',
  styleUrl: './pedigree.component.scss'
})
export class PedigreeComponent implements OnInit {
  @Input({ required: true }) goat!: Goat;
  dam?: Goat;
  sire?: Goat;
  damDam?: Goat;
  damSire?: Goat;
  sireDam?: Goat;
  sireSire?: Goat;
  constructor(private goatService: GoatService) { }
  ngOnInit() {
    this.goatService.related.subscribe(goats => {
      this.dam = goats.find(goat => goat.id === this.goat.damId);
      this.sire = goats.find(goat => goat.id === this.goat.sireId);
      this.damDam = goats.find(goat => goat.id === this.dam?.damId);
      this.damSire = goats.find(goat => goat.id === this.dam?.sireId);
      this.sireDam = goats.find(goat => goat.id === this.sire?.damId);
      this.sireSire = goats.find(goat => goat.id === this.sire?.sireId);
    }
    );
  }
}
