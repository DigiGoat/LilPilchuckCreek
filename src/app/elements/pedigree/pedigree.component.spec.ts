import { of } from 'rxjs';

import { ComponentFixture, TestBed } from '@angular/core/testing';

import { GoatService } from '../../services/goat/goat.service';
import { PedigreeComponent } from './pedigree.component';


jest.mock('../../services/goat/goat.service');
const goatService = jest.mocked(GoatService);
describe('DoesComponent', () => {
  let component: PedigreeComponent;
  let fixture: ComponentFixture<PedigreeComponent>;

  beforeEach(async () => {
    goatService.mockClear();
    await TestBed.configureTestingModule({
      declarations: [PedigreeComponent],
      providers: [GoatService],
    })
      .compileComponents();
    fixture = TestBed.createComponent(PedigreeComponent);
    component = fixture.componentInstance;
    TestBed.inject(GoatService);

    component.goat = {};

    component['goatService'] = {
      related: of([])
    } as unknown as GoatService;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
  it('should call GoatService once', () => {
    expect(goatService).toHaveBeenCalledTimes(1);
  });
  it('should match the snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
