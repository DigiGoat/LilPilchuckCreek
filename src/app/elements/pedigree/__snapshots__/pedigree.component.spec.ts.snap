// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoesComponent should match the snapshot 1`] = `
<app-pedigree
  dam="undefined"
  damDam="undefined"
  damSire="undefined"
  goat={[Function Object]}
  goatService={[Function Object]}
  sire="undefined"
  sireDam="undefined"
  sireSire="undefined"
>
  <div
    class="text-center mt-4"
  >
  </div>
</app-pedigree>
`;
